xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-api/src/main/java/org/xwiki/attachment/MoveAttachmentRequest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.attachment;
+
+import org.xwiki.model.reference.AttachmentReference;
+import org.xwiki.refactoring.job.EntityRequest;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Job request for moving an attachment to a new location. A redirection can be persisted at the old location.
+ *
+ * @version $Id$
+ * @since 14.0RC1
+ */
+@Unstable
+public class MoveAttachmentRequest extends EntityRequest
```

Comment:
I'm wondering if this should not go in xwiki-platform-refactoring-api with the other entities refactoring jobs.

Reply:
Only the request or also the corresponding job? I'm not sure it's interesting as it add yet another dependency on the refactoring module towards `attachment-api`. That's fine but that will make `refactoring` dependent on a lot of modules.